pkg/collector/tcp.go

```diff
@@ -81,9 +81,14 @@ func (cp *CollectingProcess) handleTCPClient(conn net.Conn) {
 	}()
 	defer conn.Close()
 	reader := bufio.NewReader(conn)
+	doneCh := make(chan struct{})
 	cp.wg.Add(1)
+	// We read from the connection in a separate goroutine, so we can stop immediately when
+	// cp.StopChan is closed. An alternative would be to use a read deadline, and check
+	// cp.StopChan at every iteration.
 	go func() {
 		defer cp.wg.Done()
+		defer close(doneCh)
```

Comment:
this could also be considered a bug fix. Prior to this commit, `return` statements in the `for` loop would cause this goroutine to return, but we would block on `cp.stopChan` indefinitely (so I think there would be a goroutine leak, as the parent goroutine, which invoked `handleTCPClient` would never return).

Reply:
Wouldn't it still be blocked in startTCPServer?